docs/source/queues.rst

```diff
@@ -0,0 +1,133 @@
+.. meta::
+   :description: Mergify Documentation for Queues rules
+   :keywords: mergify, queues
+
+.. _Queue rules:
+
+==============
+🔀 Queue Rules
+==============
+
+|premium plan tag|
+|beta tag|
+
+``queues_rules`` are used by the ``queue`` action. They define the conditions
+to get a queued pull request merged and the configuration of the merge
+behavior.
+
+The ``queues_rules`` and the ``queue`` action allow writing complex merge queue
+workflows that couldn't be achieved with the ``merge`` action.
+
+The merge train
+~~~~~~~~~~~~~~~
+
+With ``merge`` action, pull requests are rebased/updated and merged one by one.
+
+Instead, pull request queued with the ``queue`` action will now got embarked
+into a merge train. Mergify will create temporary pull requests to verify
+the future of the target branch can be merged according your ``queue_rules```.
+
+For example, the base branch head commit is ``abcdef12`` and you have the pull
+request ``A``, ``B`` and ``C`` embarked to merge with the ``queue`` action.
+Mergify creates the following pull requests:
+
+* ``W`` with the content of ``abcdef12`` + ``A``
+* ``X`` with the content of ``abcdef12`` + ``A`` + ``B``
+* ``Y`` with the content of ``abcdef12`` + ``A`` + ``B`` + ``C``
+
+
+Current Mergify embarks up to 5 pull requests to the merge train.
+
+Continuous Integration pipelines can run on all of them in parallel and when
+the pull request ``W`` matches the ``queue_rules``, the pull request ``A`` it
+merged and the pull request ``W`` is deleted. And then we wait for the pull
+request ``X`` to matches the ``queue_rules`` to merge the pull request ``B``.
+etc...
+
+Now, pull request ``D`` is embarked in merge train, ``B`` have been merged in
+the base branch, its head commit have moved to ``12345678``. Mergify will creates:
+
+* ``Z`` with the content of ``12345678`` + ``C`` + ``D``
+
+If anything is not handled by Mergify is merged in the base branch, the merge
+train is reset and the process start over.
+
+Working with Queue Rules
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+With a ``queue`` action, a pull request has to match two sets of conditions
+to get merged. The first set of conditions described in ``pull_request_rules``
+allows the pull request to enter in the queue when all conditions match. If
+the ``queue`` action conditions do not match anymore the pull request is
+removed from the merge queue. When a pull request is the next one to be
+merged, Mergify looks at the seconds set of conditions defined this time in
+``queue_rules`` and wait that they all match to merge the pull request.
+
+When multiple queues are defined, they are processed one after the other in the
+order they are defined in the configuration. When the first pull request in the
+first queue is waiting to be merged, the remaining queues are blocked.
+
```

Comment:
we should add something about the cases where the PR is removed from the queue: either by being merged or by not matching the conditions anymore.